src/solidlsp/ls_utils.py

```diff
@@ -173,8 +174,15 @@ def read_file(logger: LanguageServerLogger, file_path: str, encoding: str) -> st
             logger.log(f"File read '{file_path}' failed: File does not exist.", logging.ERROR)
             raise SolidLSPException(f"File read '{file_path}' failed: File does not exist.")
         try:
-            with open(file_path, encoding=encoding) as inp_file:
-                return inp_file.read()
+            try:
+                with open(file_path, encoding=encoding) as inp_file:
+                    return inp_file.read()
+            except UnicodeDecodeError as ude:
+                results = from_path(file_path)
+                match = results.best()
+                if match:
+                    return match.raw.decode(match.encoding)
```

Comment:
Let's log a warning here, e.g.

    log.warning(f"Could not decode {file_path} with encoding='{encoding}'; using best match '{match.encoding}' instead")